### PR TITLE
Shortcut improvements

### LINKED ISSUE #3583: Shortcut improvements
### Description of the Issue
In Shortcut mapper: 

-  we don't know which  category the shortcut belongs to
- we don't know which plugin the shortcut belongs to
- we don't see every shortcuts mapped in scintilla view.

### Expected Behavior

- add a 3rd column in the main menu that contains a category ("File", "Edit", "Format", "Search" ....)
- add a 3rd column in the plugin commands that contains the plugin name
- in the scintilla panel, when they are multiple mapped shortcut, print all of them with "or" separator.

Checkout the screenshots 
before pull request 
nppbeforeshortcut

after pull request
nppaftershortcut

### PR DESCRIPTION
**Pull request by cmeriaux**
Fixes #3583

Shorcut mapper - main panel : new colums that show the category of the shortcut
Shorcut mapper - plugin panel : new colums that show the plugin name that the shortcut belongs to
Shorcut mapper - scintilla panel : it shows every shortcuts configured for one command

### CODE REVIEW

**Comment by donho** [comment]
@cmeriaux  Are you sure that you have created a new branch from Notepad++'s master?
The branch is still named `cmeriaux-shortcutImprovements`

File: PowerEditor/src/WinControls/shortcut/shortcut.cpp [hunk 1]
```diff
@@ -187,7 +187,14 @@ void Shortcut::setName(const TCHAR * name) {
 }
 
 generic_string ScintillaKeyMap::toString() const {
```
Comment (donho) [inline PowerEditor/src/WinControls/shortcut/shortcut.cpp]: Bad style, please check https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/CONTRIBUTING.md#coding-style

File: PowerEditor/src/WinControls/shortcut/shortcut.cpp [hunk 2]
```diff
@@ -187,7 +187,14 @@ void Shortcut::setName(const TCHAR * name) {
 }
 
 generic_string ScintillaKeyMap::toString() const {
-	return toString(0);
+	generic_string sc = TEXT("");
+	size_t nbCombos = getSize();
+	for (size_t combo = 0; combo < nbCombos; ++combo){
```
Comment (donho) [inline PowerEditor/src/WinControls/shortcut/shortcut.cpp]: Bad style;

File: PowerEditor/src/WinControls/shortcut/shortcut.cpp [hunk 3]
```diff
@@ -948,3 +955,30 @@ INT_PTR CALLBACK ScintillaKeyMap::run_dlgProc(UINT Message, WPARAM wParam, LPARA
 
 	//return FALSE;
 }
+
+CommandShortcut::CommandShortcut(Shortcut sc, long id) :	Shortcut(sc), _id(id) {
```
Comment (donho) [inline PowerEditor/src/WinControls/shortcut/shortcut.cpp]: Bad style.

**Comment by cmeriaux** [comment]
@donho actually I didn't create a new branch. I rebased the existing branch onto an updated master then pushed again.
Do you want a new PR for the two bad styles mentionned ?

**Pull request closed without merging**